### ISSUE TITLE
Examine integration with ckan metadata

### ISSUE THREAD

**Issue by cgruber** [oharaandrew314/TinkerTime#228]
https://github.com/KSP-CKAN/CKAN maintains metadata for modules, including inter-module dependency information.  Strictly speaking it a quasi competitor, but is doing good work (it seems) around unifying inter-mod dependencies, so it might be worth using as a full or partial underpinning of the metadata.

**Comment by cgruber** [comment]
In my understanding they have a giithub-hosted metadata repo containing json files, so it's that that I am proposing using, not ckan itself.

**Comment by oharaandrew314** [comment]
Yeah, I've heard of CKAN before, but I've never seen people actually writing specs for their mods.  Since CKAN can generate the spec itself, then the metadata repo may be useful.

**Comment by oharaandrew314** [comment]
Wow, the [CKAN-meta](https://github.com/KSP-CKAN/CKAN-meta) repo is incredible.  I'll probably drop #153 in favor of this.  I think I'll make this current milestone v1.4, and make CKAN integration v2.0.  It could change everything.  Rather than making users paste links from KS, GH, and Curse, I can add a GUI for them to search the repo and add mods that way.  I'll keep the manual mod zip adding for the case that the repo doesn't have data for a particular mod.

**Comment by oharaandrew314** [comment]
I guess TinkerTime will essentially become a CKAN client, but running on the Java platform? :confused:

**Comment by oharaandrew314** [comment]
I guess TinkerTime will essentially become a CKAN client. What's the point in even continuing development after v1.4?  :confused:

**Comment by cgruber** [comment]
Because the ckan GUI client dies on the mac, has some real issues because
of mono, and tinker-time can still be an amazing alternative GUI that does
thigns the way you want them?

**Comment by oharaandrew314** [comment]
Yeah, I guess that's good incentive!

**Comment by oharaandrew314** [comment]
According to [iulianov](https://github.com/iulianov), CKAN-meta doesn't always offer the latest releases of the mods.  This may make integration more complicated.

I think I can still have full reliance on CKAN, but then I just use a crawler to check the resource pages for updates.

**Comment by cgruber** [comment]
It's pretty close.  It offers whatever the mod publishers add to the ckan
net. That's probably good enough. Though the added up to date check is
likely a good added value.
